test/external-entity-references-melinda.spec.js

```diff
@@ -0,0 +1,166 @@
+chai.use(chaiAsPromised);
+
+const endpoint = 'http://melinda.kansalliskirjasto.fi:210/fin01';
+const queryParam = '?operation=searchRetrieve&maximumRecords=2&version=1&query=rec.id=';
+
+const prefixPattern = /^\(FOOBAR\)/;
+const fields = {
+	773: ['w'],
+	833: ['w', 'p']
+};
+
+describe('external-entity-references-melinda', () => {
+	afterEach(() => {
+		testContext.default.__ResetDependency__('fetch');
+	});
+
+	it('Creates a validator', async () => {
+		const validator = await testContext.default({endpoint, prefixPattern, fields});
+
+		expect(validator)
+			.to.be.an('object')
+			.that.has.any.keys('description', 'validate');
+
+		expect(validator.description).to.be.a('string');
+		expect(validator.validate).to.be.a('function');
+	});
+
+	it('Throws an error when prefixPattern not provided', async () => {
+		const validator = await testContext.default({endpoint, prefixPattern, fields});
+		await expect(validator.validate()).to.be.rejectedWith(Error, 'Cannot read property \'fields\' of undefined');
+	});
+
+	describe('#validate', () => {
+		it('Finds prefixPattern on record and removes it', async () => {
+			const mock = fetchMock.sandbox();
+			mock.get(`${endpoint}${queryParam}5000`, fixture5000);
+			mock.get(`${endpoint}${queryParam}9550`, fixture9550);
+
+			testContext.default.__Rewire__('fetch', mock);
+			const validator = await testContext.default({endpoint, prefixPattern, fields});
+
+			const record = new MarcRecord({
+				fields: [
+					{
+						tag: '001',
+						value: '123456'
+					},
+					{
+						tag: '035',
+						subfields: [
+							{
+								code: 'a',
+								value: '(FI-MELINDA)123456'
+							}
+						]
+					},
+					{
+						tag: '773',
+						subfields: [
+							{
+								code: 'w',
+								value: '(FOOBAR)5000'
+							}
+						]
+					},
+					{
+						tag: '833',
+						subfields: [
+							{
+								code: 'p',
+								value: '(FOOBAR)9550'
+							},
+							{
+								code: 'c',
+								value: '(FI-MELINDA)8850'
+							}
+						]
+					}
+				]
+			});
+			const result = await validator.validate(record);
+
+			expect(result).to.eql({valid: true});
+		});
+	});
+
+	describe('#validate', () => {
+		it('Finds no matching prefixPattern on record', async () => {
+			const mock = fetchMock.sandbox();
+
+			mock.get(`${endpoint}5000`, fixture5000);
+
+			testContext.default.__Rewire__('fetch', mock);
+			const validator = await testContext.default({endpoint, prefixPattern, fields});
+
+			const record = new MarcRecord({
+				fields: [
+					{
+						tag: '001',
+						value: '123456'
+					},
+					{
+						tag: '035',
+						subfields: [
+							{
+								code: 'a',
+								value: '(FI-MELINDA)123456'
+							}
+						]
+					},
+					{
+						tag: '773',
+						subfields: [
+							{
+								code: 'w',
+								value: '(FI-MELINDA)123456'
+							}
+						]
+					}
+				]
+			});
+			const result = await validator.validate(record);
+
+			expect(result).to.eql({valid: false});
```

Comment:
Should be `{valid: true}` since finding to values is ok.

Reply:
fixed